Spec/RestClient.swift

```diff
@@ -315,124 +315,176 @@ class RestClient: QuickSpec {
                         }
-                        
-                        guard let currentTokenDetails = tokenDetails else {
-                            XCTFail("expected tokenDetails not to be nil when error is nil")
-                            done()
-                            return
-                        }
-                        
-                        let options = AblyTests.clientOptions()
-                        options.key = client.options.key
+                    }
+                }
 
-                        // Expired token
-                        options.tokenDetails = ARTTokenDetails(token: currentTokenDetails.token, expires: currentTokenDetails.expires!.dateByAddingTimeInterval(testTimeout), issued: currentTokenDetails.issued, capability: currentTokenDetails.capability, clientId: currentTokenDetails.clientId)
+                // RSC15b
+                pending("applies when the default rest.ably.io endpoint is being used") {
+                    let options = ARTClientOptions(key: "xxxx:xxxx")
+                    let client = ARTRest(options: options)
+                    client.httpExecutor = testHTTPExecutor
+                    testHTTPExecutor.http = MockHTTP(network: .HostUnreachable)
+                    let channel = client.channels.get("test")
 
-                        let rest = ARTRest(options: options)
-                        rest.httpExecutor = testHTTPExecutor
+                    testHTTPExecutor.afterRequest = { _ in
+                        if testHTTPExecutor.requests.count == 2 {
+                            // Stop
+                            testHTTPExecutor.http = nil
+                        }
+                    }
 
-                        // Delay for token expiration
-                        delay(tokenParams.ttl) {
-                            // [40140, 40150) - token expired and will not recover because authUrl is invalid
-                            publishTestMessage(rest) { error in
-                                guard let errorCode = testHTTPExecutor.responses.first?.allHeaderFields["X-Ably-ErrorCode"] as? String else {
-                                    fail("expected X-Ably-ErrorCode header in request")
-                                    return
-                                }
-                                expect(Int(errorCode)).to(beGreaterThanOrEqualTo(40140))
-                                expect(Int(errorCode)).to(beLessThan(40150))
-                                expect(error).to(beNil())
-                                expect(rest.auth.tokenDetails!.token).toNot(equal(currentTokenDetails.token))
-                                done()
-                            }
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish(nil, data: "nil") { _ in
+                            done()
                         }
                     }
+
+                    expect(testHTTPExecutor.requests).to(haveCount(2))
```

Comment:
Why do we expect only two requests?  It should try all the fallback hosts surely, well at least up until `httpMaxRetryCount`